pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix

```diff
@@ -22,7 +22,7 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
     wrapProgram $out/bin/git-remote-hg \
-      --prefix PYTHONPATH : "$(echo ${mercurial}/lib/python*/site-packages):$(echo ${mercurial.python}/lib/python*/site-packages)${stdenv.lib.concatMapStrings (x: ":$(echo ${x}/lib/python*/site-packages)") mercurial.pythonPackages or []}"
+      --set PYTHONPATH "$(echo ${mercurial}/lib/python*/site-packages):$(echo ${mercurial.python}/lib/python*/site-packages)${stdenv.lib.concatMapStrings (x: ":$(echo ${x}/lib/python*/site-packages)") mercurial.pythonPackages or []}"
```

Comment:
`buildPython*` should be able to wrap this when you put the runtime deps in `propagatedBuildInputs`.

Reply:
The problem is, `setup.py` is currently broken, because it tries to fetch the version git a git tag and fails. We can patch it though.

Reply:
> buildPython* should be able to wrap this when you put the runtime deps in propagatedBuildInputs.

This package depends on `mercurial`, which is built with `buildPythonApplication`. Will it still work?

Reply:
> The problem is, `setup.py` is currently broken, because [it tries to fetch the version from a git tag](https://github.com/mnauw/git-remote-hg/blob/f050de1bcc3f37f5572035a63c85aba60f7b9088/setup.py#L9) and fails. We can patch it though.

Yeah they're many packages that take a much better approach that they could have used as an example. Patching and submitting it upstream would be preferred. 

But if it's too much of a hassle, you can use `format = "other";` and just use the `Makefile` as normally.

> https://nixos.org/nixpkgs/manual/#python  9.14.2.2.1.1. buildPythonPackage parameters


> This package depends on mercurial, which is built with buildPythonApplication. Will it still work?

Think so.